### PR TITLE
feat: expand and visualize auth states

### PR DESCRIPTION
**Pull request by sebastianvitterso**
## Why is this pull request needed?
@eaksnes and I discussed that this was needed.

## What does this pull request change?
This PR expands and visualizes the authentication states in which the application can be.

## Issues related to this change:
None.

### CODE REVIEW

File: web/src/App.tsx [hunk 1]
```diff
@@ -1,8 +1,47 @@
...
+  }
+
+  if (hasAuthConfig && !token) {
+    return (
+      <CenterContainer>
+        <Button onClick={login}>Log in</Button>
+      </CenterContainer>
+    )
+  }
+
```
Comment (sebastianvitterso) [inline web/src/App.tsx:44]: I welcome discussion on whether or not this code should lie in `App.tsx`.

File: web/src/auth.ts [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
+import { TAuthConfig } from 'react-oauth2-code-pkce'
+
+export const authConfig: TAuthConfig = {
   clientId: process.env.REACT_APP_AUTH_CLIENT_ID || '',
   authorizationEndpoint: process.env.REACT_APP_AUTH_ENDPOINT || '',
   tokenEndpoint: process.env.REACT_APP_TOKEN_ENDPOINT || '',
   scope: process.env.REACT_APP_AUTH_SCOPE || '',
   redirectUri: process.env.REACT_APP_AUTH_REDIRECT_URI || '',
   logoutEndpoint: process.env.REACT_APP_LOGOUT_ENDPOINT || '',
+  autoLogin: false,
```
Comment (sebastianvitterso) [inline web/src/auth.ts:10]: If y'all thing we should re-enable `autoLogin`, I will.

File: web/src/App.tsx (hunk 1 above)
Comment (eoaksnes) [reply web/src/App.tsx:44]: Could also be added to App, not sure what is the best

File: web/src/auth.ts (hunk 2 above)
Comment (eoaksnes) [reply web/src/auth.ts:10]: roger

**Review by eoaksnes — APPROVED** [review]

File: web/src/App.tsx (hunk 1 above)
Comment (sebastianvitterso) [reply web/src/App.tsx:44]: I don't think I understand, @eoaksnes..

**Pull request merged**